src/humanloop/otel/exporter.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import typing
+from queue import Queue
+from threading import Thread
+from typing import Any, Optional
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+
+from humanloop.otel.constants import HL_FILE_OT_KEY, HL_LOG_OT_KEY, HL_OT_EMPTY_VALUE, HL_TRACE_METADATA_KEY
+from humanloop.otel.helpers import is_humanloop_span, read_from_opentelemetry_span
+from humanloop.requests.flow_kernel_request import FlowKernelRequestParams
+from humanloop.requests.prompt_kernel_request import PromptKernelRequestParams
+
+if typing.TYPE_CHECKING:
+    from humanloop.base_client import BaseHumanloop
+
+
+logger = logging.getLogger("humanloop.sdk")
+
+
+class HumanloopSpanExporter(SpanExporter):
+    """Upload Spans created by SDK decorators to Humanloop.
+
+    Spans not created by Humanloop SDK decorators will be ignored.
+    """
+
+    DEFAULT_NUMBER_THREADS = 4
+
+    def __init__(
+        self,
+        client: "BaseHumanloop",
+        worker_threads: Optional[int] = None,
+    ) -> None:
+        super().__init__()
+        self._client = client
+        self._uploaded_log_ids: dict[
+            str, str
+        ] = {}  # Uploaded spans translate to a Log on Humanloop. The IDs are required to link Logs in a Flow Trace
+        self._upload_queue: Queue = Queue()  # Work queue for the threads uploading the spans
+        self._threads: list[Thread] = [
+            Thread(target=self._do_work, daemon=True) for _ in range(worker_threads or self.DEFAULT_NUMBER_THREADS)
+        ]
+        self._shutdown: bool = (
+            False  # Signals threads no more work will arrive and they should wind down if the queue is empty
+        )
+        for thread in self._threads:
+            thread.start()
+
+    def export(self, spans: trace.Sequence[ReadableSpan]) -> SpanExportResult:
```

Comment:
Want doc strings on all these methods

Reply:
The methods are inherited from Otel's SpanExporter and already have docstrings
<img width="582" alt="image" src="https://github.com/user-attachments/assets/89f2b5fd-cd6b-4619-9170-62a72ade1c7a">